### DISCUSSION TITLE
👨‍💻☎ WinUI Community Call: August 17, 2022

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/microsoft-ui-xaml#7375] [Announcements]
### https://www.youtube.com/watch?v=W7DCJOmawrs

# Details
Date: August 17, 2022
Time: 16:00-17:00 UTC (9:00-10:00am Pacific)

Anyone and everyone is welcome - no pre-registration is required.
This will be an informal interactive live stream directly with members of our engineering team.

# Format
The community call is a call among the WinUI team that is live-streamed onto YouTube. We present on new updates, share information, welcome guests, and answer your questions. In this month's call we'll share the latest info about recent and upcoming WinUI & WinAppSDK releases and joined by several team members to discuss features across the platforms. These include the WinAppSDK AppLifecycle APIs for WinUI 3 applications, an overview of WebView2 for WinUI 2.8 & WinUI 3 apps, and the new Content APIs introduced in WinAppSDK 1.2-experimental.

### **Agenda**
- Intro
- @dhoehna: App Lifecycle in WinUI 3 applications
- @krschau: WebView2 in WinUI 2 & WinUI 3
- @stevewri: 1.2-experimental: Content APIs & Samples
- Latest info on WinAppSDK & WinUI releases
- Q&A

### **Q&A Code of Conduct**
- To help ensure Q&A remains beneficial for the majority of the call's developer audience, please consider whether your question could be framed so that it might be more broadly relevant to our larger developer audience by ensuring it is reasonably concise, broadly beneficial, well-suited for the fast-paced live Q&A format, or able to spark open ended discussion.  
- Please understand some questions are not well-suited for the fast-paced live Q&A format and should be filed as standalone [Q&A](https://github.com/microsoft/microsoft-ui-xaml/discussions/new?category=q-a&title=%5BYour+Question+Here%5D) discussions where they can be most appropriately handled. Excluding presentation topic requests, this generally includes requests for dedicated follow-ups or technically complex questions that require investigative time or expertise not included in this month's guest list.

### Leave us your questions in the comments on this issue or live in the YouTube chat during the stream!

**Comment by wbokkers** [comment]
After updating our app from WinAppSDK 1.0.4 to 1.1.2, our users report lots of crashes. Especially with drag & drop and adding items to listviews. After switching back to 1.0.4 users are much happier. It's much more stable.
* Are there any known regressions in the 1.1.x version?
* How is the team tackling these stability regressions?
* When will it finally be possible to get diagnosis from users reporting 'silent' crashes => crashes I currently have no way to handle.

**Reply by Scottj1s** [reply]
The Drag Drop crashes will be fixed in WinAppSDK 1.2

**Comment by TRadigk** [comment]
Is https://portal.productboard.com/winappsdk/1-windows-app-sdk/tabs/2-planned abandoned?

**Comment by arpadbarta** [comment]
### Stability and basic features

- It seems that even after the 1.0 release there are several regressions when it comes to stability, is there a plan to improve this?
- There are several long outstanding issues logged for basic functionalities such as setting a property value from XAML ex: https://github.com/microsoft/microsoft-ui-xaml/issues/3689.
Is it a valid concern that not addressing these issues will lead to sever distrust in the WinUI platform and MSs capability to make it viable?

**Reply by arpadbarta** [reply]
Was hoping that by putting these questions forward, allows the team to share some information about the topics and calm things down.
I do not know why some of these old issues are still open, it is a bit frustrating, ignoring them, and the feedback about it will certainly not help.

**Reply by pjmlp** [reply]
Look at the buzz around MAUI and Blazor, PWAs, Azure support for .NET and then the steady increase of WinUI issues.

MAUI might depend on WinUI for Windows workloads, and not even them were important enough as customer to make Win2D migration story happen, they are using SkiaSharp for their 2D hardware rendering.

In MFC and WPF accessing accelerated 2D rendering is a couple of calls away, here it seems most of the team aren't even aware of what the frameworks they are supposed to replace are capable of, in features and Visual Studio tooling.

After all reboots since Windows 8, I expect yet another one by the Windows 12 timeframe.

Just keep using Forms, WPF, MFC, if you want modern C++ GUI development then C++ Builder or Qt, or just plain Web/hybrid.

**Comment by HEIC-to-JPEG-Dev** [comment]
When will WinUI get a xaml designer as hot reload doesn’t yet support all xaml features, like inline rowdefinitions, and hunting for properties without the grouping and exposure of the properties window means we have to keep referring to an alphabetical list in the documentation.

**Reply by llothar** [reply]
I would expand on this and https://github.com/microsoft/microsoft-ui-xaml/issues/7606 and ask when will this framework so open and documented that other people (like Rafael Rivera from the Rust team) can write it's own xaml designer independent from the Microsoft tool chain.

**Comment by ibiller** [comment]
When will all Microsoft supported nuget packages support unpackaged WinUI 3 apps? Things like drop shadows don't work outside of VS because Win2D doesn't support unpackaged apps, and Win2D looks like a dead project.
When will documentation get updated? There are still a lot of references everywhere to project reunion and version numbers < 1.0. The documentation on MSDN pretty much redirects to UWP. Its August, we're getting close to circling back to November for an anniversary, it doesn't look good.

**Reply by ibiller** [reply]
I guess this was sort of dodged in the call ...
Here's a more concrete example:
I want dropshadows in WinUI3. They don't really seem to work via normal routes, there's no real documentation on how it should work, no clean examples. The only viable method appears to be via WCT. I use WCT, it works. Great! I go to deploy unpackaged, it crashes. I look at WCT issues on github: https://github.com/CommunityToolkit/WindowsCommunityToolkit/issues/4406
in there they say there's nothing they can do because it traces to the Win2D project. They link open issue https://github.com/microsoft/Win2D/issues/831 from May 2021. It sits with a ton of other issues untouched.
Then if you look at the Win2D github at https://github.com/microsoft/Win2D, maybe they really are still working on transitioning to "Reunion", without any commits in 4 months... Is the right answer to open yet another issue on github to go ignored?

Is it really the greater community's job to report out of date documentation everywhere? Does nobody own documenting? I'm using Win2D to make my point but its in the MSDN docs too. I'm definitely upset that there's seemingly no traction on fixing this, and I only scratch the surface on the range of features that one can utilize. We need unpackaged support on all Microsoft Nuget packages!

**Reply by pjmlp** [reply]
Back to Win32 and Direct2D, or Skia, that is hopeless to expect anything else, really frustrating.

**Comment by Marv51** [comment]
How will open-source WinUI work going forward? I believe you said, that no WinUI 2.9 release is currently planned. Now, WinUI 3 is really not directly based on the open-source WinUI code you publish here.

So will accepting pull-requests and viewing the source in this repo, be even more difficult now? 
Are you planning to keep the control layer of WinUI still open source?

**Comment by nikolayvpavlov** [comment]
Are there plans to improve the integration with the shell of the operating system?  For example, integration with task bar (ITaskbarlist3)? Currently, too many features require interop and using Win32 User Apis.  This is definitely inferior to WinForms.  WinUI must be extended to support what WinForms does for window management.

Are there plans to improve the facilities for Interop, especially with Windowing, Messaging and the Shell?  Trapping windows messages (for example, WM_COMMAND) should not require subclassing the main window via interop and going through arcane interop stuff, that will take us back to the dark ages of Win32.  

Are there plans to support WinForms / Win32 control islands?  While there are [well-known issues](https://docs.microsoft.com/en-us/archive/blogs/dwayneneed/mitigating-airspace-issues-in-wpf-applications), achieving the functional level of WPF will be more than sufficient for basic stuff.  The foundation is seemingly there with WebView2.

**Reply by llothar** [reply]
The dark ages of Win32 feeling much better than ther bright days of WinUI3. 

At least Win32 was documentend so well and simpler to use and understand even if it was pretty rough to program.

But WinUI3 is nothing i can hand over to any Junior programmer in our company.

**Reply by dhoehna** [reply]
I haven't done any major WinUI3 work I might be missing the bigger picture.  

I'm curious, why can't you give a WinUI3 app to a junior developer?

Not saying anyone is wrong, I am just curious.

**Reply by MEK3DK** [reply]
@dhoehna you can't because e.g. there are to many basic features missing atm, which you have to workaround, like the example mentioned in the top where you cannot set the Title property on a window.
With e.g. a WPF app you can just go ahead and create a new project and compile and you have a program. With WinUI3 you have to jump through a few hoops and the documentation will not help you much (yet).
There are lots of other examples/reasons, but basically WinUI3 is just not there yet.

**Reply by dhoehna** [reply]
@MEK3DK Interesting.   Thank you for sharing your thoughts.

About the documentation to start a winui 3 project.  Is there an issue with the documentation here:  [Create your first Winui 3](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app)  Seemed straight forward enough.  I was able to make my first Winui3 project.

> where you cannot set the Title property on a window.
I'm sorry.  I must be missing something.  Does `m_MainWindow.Title = "My Awesome Window"` not do something that is needed?  Works pretty well for setting the title.

**Reply by MEK3DK** [reply]
@dhoehna No those docs are likely correct, and it seems to get better every version. I have not tried all the scenarios, but people seems to struggle especially to the deployment part, thus a junior developer would likely also.
The Title thing was just an example, 2600 other issues to pick from :-) but see here for more details: #3689

**Reply by dhoehna** [reply]
@MEK3DK Oh.  Wow.  Those are a lot of issues.

I was not aware of these issues when I made my first comment.  This changes things.

> thus a junior developer would likely also.
I agree.

**Reply by llothar** [reply]
And if your projects are C++ there is a whole universe of other bugs. The worst tooling experience in the last 20 years is some of it.
I can't use code insight and debugging is also unuseable bad. If you are debugging for 5 years it might be just inconvinient but for newbies its fatal.
 
And than there is the problem that Junior Programmers might have no windows experience at all. All the current documentation assumes that you know the basics from WPF or UWP already even if the knowledge is only partial reuseable. Take a junior who comes fresh from university and only knows the basic language. You can't even give them a book or a print out to read as there is none and the developer pages are not PDF printable.

And finally as you found out by know there are thousands of issues without solutions.

**Reply by pjmlp** [reply]
It is so sad that MFC happens to have better tooling than C++/WinRT. Hopefully the management that took the call to deprecate C++/CX and bring us to the stone age before .NET was invented, is no longer at Microsoft.

**Comment by AathifMahir** [comment]
Any News on Native Splash Screen Support for WinUI 3 Apps Like UWP? 

Currently different WinUI 3 Apps Goes with Custom Solution for Splash Screen that Makes WinUI 3 Apps Not Consistent and Bad At Initial User Experience

**Reply by Jdaslepre** [reply]
I'd also love to see a splash screen instead of looking at a black void

**Comment by Jdaslepre** [comment]
When will the window resizing issue be fixed? When resizing a window, the content's resizing is delayed and shows a black or white background until it's ready. Also when resizing, the window will flicker with the content not being visible at all for a few milliseconds

**Reply by TRadigk** [reply]
@CalcuIator  Do you have an issue# for that? I've never seen anything like this. Neither with or without Mica...

**Reply by Jdaslepre** [reply]
> @CalcuIator  Do you have an issue# for that? I've never seen anything like this. Neither with or without Mica...

The issue should be visible here https://github.com/microsoft/WindowsAppSDK/issues/2852

**Reply by TRadigk** [reply]
Thanks, nice video. It might be important to know that this is only that much visible when the **dialog** window has **dimmed** the background of the window. Your question made it seem like every window has this effect regardless of its content.

**Reply by Jdaslepre** [reply]
> Thanks, nice video. It might be important to know that this is only that much visible when the **dialog** window has **dimmed** the background of the window. Your question made it seem like every window has this effect regardless of its content.

You can still notice the content's resizing delay when there's no dialog :(
https://user-images.githubusercontent.com/104047615/184503111-8d311f66-4818-4028-9737-01f966008559.mp4

**Reply by sudongg** [reply]
> Thanks, nice video. It might be important to know that this is only that much visible when the **dialog** window has **dimmed** the background of the window. Your question made it seem like every window has this effect regardless of its content.

The problem has nothing to do with having a dialog, it happens whenever the window is zoomed.

**Comment by sabin26** [comment]
When will the issue of spinning cursor (busy) during WinUI3 application startup be fixed ?

**Comment by artkat** [comment]
Hate to bring this up again, but when can we expect bug #5221 (App.UnhandledException doesn't catch exceptions from other threads) to be resolved?

**Reply by Scottj1s** [reply]
The App.UnhandledException crashes will be fixed in WinAppSDK 1.2

**Reply by artkat** [reply]
So that will be around the end of the year, if all goes well...??

**Reply by Scottj1s** [reply]
End of September, for Preview 2, if all goes well

**Reply by wbokkers** [reply]
Can Preview 2 be use in production? If it is not production-ready, I can't use it in our app.

**Comment by MEK3DK** [comment]
Are you going to address the lack of a designer in Visual Studio somehow? Currently people who wants to try WinUI3 some times spend hours trying to figure out why their designer does not work, just to realize in the end that there is no designer: [WinUI3 designer missing](https://github.com/microsoft/microsoft-ui-xaml/issues/5917)
At least show a message that there is no designer.

**Comment by terrycox** [comment]
What's the status of support for PrintManager API? 
(https://portal.productboard.com/winappsdk/1-windows-app-sdk/c/50-support-printmanager-api)

**Comment by pubtom** [comment]
1.) When will you fix the unable to Interact with WebView in WinUI version > 2.5 issue?
WinUI 2.8 still has this issue! This is a blocking issue: WinUI2 is unusable since the 2.6 version if your app uses a WebView control.
I forced to use the WinUI 2.5 version because of this issue. Highly waiting for the fix!
see https://github.com/microsoft/microsoft-ui-xaml/issues/5299

2.) When will WinUI 2.8 version of WebView2 run on UWP apps on Xbox?
see https://github.com/MicrosoftEdge/WebView2Feedback/issues/2155
